smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java

```diff
@@ -136,7 +136,14 @@ private void writePager() {
             writer.write("let hasNext = true;");
             writer.write("let page: $L;", outputTypeName);
             writer.openBlock("while (hasNext) {", "}", () -> {
-                writer.write("input[$S] = token;", inputTokenName);
+                if (inputTokenName.contains(".")) {
+                    // Smithy allows one level indexing (ex. 'bucket.inputToken').
```

Comment:
We allow any number of nesting

Reply:
Addressed - destructured the path into an array index. Should support arbitrary length